aws/resource_aws_dynamodb_table_test.go

```diff
@@ -103,7 +392,33 @@ func TestAccAWSDynamoDbTable_gsiUpdate(t *testing.T) {
 				),
 			},
 			{
-				Config: testAccAWSDynamoDbConfigGsiUpdated(name),
+				Config: testAccAWSDynamoDbConfigGsiUpdatedOtherAttributes(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckInitialAWSDynamoDbTableExists("aws_dynamodb_table.test", &conf),
+				),
+			},
+		},
+	})
+}
+
+// https://github.com/terraform-providers/terraform-provider-aws/issues/566
+func TestAccAWSDynamoDbTable_gsiUpdateNonKeyAttributes(t *testing.T) {
+	var conf dynamodb.DescribeTableOutput
+	name := acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSDynamoDbTableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSDynamoDbConfigGsiUpdatedOtherAttributes(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckInitialAWSDynamoDbTableExists("aws_dynamodb_table.test", &conf),
+				),
+			},
+			{
+				Config: testAccAWSDynamoDbConfigGsiUpdatedNonKeyAttributes(name),
 				Check: resource.ComposeTestCheckFunc(
```

Comment:
Should we have additional attribute checks in here and after the update?

```
resource.TestCheckResourceAttr("aws_dynamodb_table.test", "global_secondary_index.####.non_key_attributes.#", "1"),
resource.TestCheckResourceAttr("aws_dynamodb_table.test", "global_secondary_index.####.non_key_attributes.0", "RandomAttribute"),
```

and

```
resource.TestCheckResourceAttr("aws_dynamodb_table.test", "global_secondary_index.####.non_key_attributes.#", "2"),
resource.TestCheckResourceAttr("aws_dynamodb_table.test", "global_secondary_index.####.non_key_attributes.0", "RandomAttribute"),
resource.TestCheckResourceAttr("aws_dynamodb_table.test", "global_secondary_index.####.non_key_attributes.1", "AnotherAttribute"),
```